# <center>R<sup>2</sup> Score</center>

In [2]:
from sklearn.metrics import r2_score

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LinearRegression
from statsmodels.compat import lzip
import statsmodels.stats.api as sms
from statsmodels.stats.outliers_influence import variance_inflation_factor as vif

%matplotlib inline

import warnings
warnings.filterwarnings("ignore")

In [4]:
df = pd.read_csv("advertising.csv")
df.head()

Unnamed: 0     TV  radio  newspaper  sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

In [5]:
X = df.drop(columns=['Unnamed: 0', 'sales'])
y = df['sales']

In [6]:
lr = LinearRegression()
lr.fit(X, y)

LinearRegression()

In [7]:
y_pred = lr.predict(X)

### R2 Score Calculation

In [9]:
r2_score(y, y_pred)

0.8972106381789522

The feature X can explain 89.72% variance in the target (y).